### ISSUE TITLE
Fail to connect to track servers after upgrade from 0256508 to 6b27e14

### ISSUE THREAD

**Issue by schemacs** [anacrolix/torrent#25]
Before upgrading, the code works, but break after upgrade as mentioned in title, I have logged the announce response:

    2015/08/14 20:05:22 udp.go:127: binary.Read:&{[0 0 6 153 0 0 0 5 0 0 0 2 10 10 41 1 133 142 10 10 41 1 152 222 10 10 41 1 154 199 10 10 41 1 233 229 10 10 41 65 195 87 10 10 41 1 222 74 10 10 41 1 172 1] 12 -1}
    client.go:2438: big: error announcing to udp://10.10.41.65:6969: error announcing: binary.Read: invalid type *util.CompactPeer

This is the related change for `CompactPeer` struct in util/types.go:

```golang
 type CompactPeer struct {
-       IP   [4]byte
+       IP   net.IP
        Port uint16
 }
 ```

The problem is not resolved even after DisableIPV6 is set to true in `client`.

**Comment by anacrolix** [comment]
Thanks for catching this. I'll issue a fix shortly.

**Comment by anacrolix** [comment]
Should be fixed by 1db00a54f000b9527ad3b97681b3c67764944156. I added some new tests, and improved the representation of compact peers. Let me know how it goes for you.
